src/app/reports/general-report/general-report.component.ts

```diff
@@ -0,0 +1,90 @@
+import {Component, OnInit} from '@angular/core';
+import GeneralReport from "../general-report.model";
+import {ReportService} from "../report.service";
+import {FormBuilder, FormControl, Validators} from "@angular/forms";
+import {ChartConfiguration} from "chart.js";
+import DailyReport from "../daily-report.model";
+import {jsPDF} from 'jspdf';
+
+@Component({
+  selector: 'app-general-report',
+  templateUrl: './general-report.component.html',
+  styleUrl: './general-report.component.css'
+})
+export class GeneralReportComponent implements OnInit {
+
+  data?: GeneralReport;
+  form = this.formBuilder.group({
+    startDate: new FormControl(new Date(), [Validators.required]),
+    endDate: new FormControl(new Date(), [Validators.required])
+  });
+
+  // Chart setup
+  lineChartLegend = true;
+  lineChartData: ChartConfiguration<'line'>['data'];
+  lineChartOptions: ChartConfiguration<'line'>['options'] = {
+    responsive: false
+  };
+
+  constructor(
+    private reportService: ReportService,
+    private formBuilder: FormBuilder
+  ) {
+  }
+
+  ngOnInit(): void {
+    this.getGeneralReport(new Date(), new Date());
+  }
+
+  getGeneralReport(start: Date, end: Date) {
+    this.reportService.getGeneralReport(start, end)
+      .subscribe(result => {
+        this.data = result;
+        this.loadLineChartData(result.dailyReports);
+      });
+  }
+
+  onSubmit() {
+    if (!this.form.valid
+      || !this.form.value.startDate
+      || !this.form.value.endDate) return;
+
+    this.getGeneralReport(
+      this.form.value.startDate,
+      this.form.value.endDate
+    );
+  }
+
+  loadLineChartData(dailyReports: DailyReport[]) {
+    console.log(dailyReports);
+
+    const dates: string[] = [], profits: number[] = [], reservationCounts: number[] = [];
+    dailyReports.forEach(report => {
+      dates.push(report.date.toISOString().slice(0, 10));
```

Comment:
There is a bug that will display a date in the general report chart as the day before if the user is in a timezone that has a positive UTC offset. Printing the date in the Swedish locale will output the date in the user's timezone in ISO 8601 format. 
Source: https://stackoverflow.com/a/60368477
```suggestion
      dates.push(report.date.toLocaleDateString("sv"));
```